ReactWindows/ReactNative.Shared/Views/View/ReactViewManager.cs

```diff
@@ -92,6 +99,192 @@ protected override BorderedCanvas CreateViewInstance(ThemedReactContext reactCon
+                            {
+                                { "name", file.Name },
+                                { "size", props.Size },
+                                { "type", type },
+                                { "uri", path }
+                            });
+
+                            items.Add(new JObject
+                            {
+                                { "kind", "file" },
+                                { "type", type }
+                            });
+
+                            types.Add(type);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        System.Diagnostics.Debug.WriteLine("GetDataTransferInfo: " + ex);
+                    }
+                }
+            }
+
+            return new JObject
+            {
+                { "files", files },
+                { "items", items },
+                { "types", types }
+            };
+        }
+
+        private string GetAccessToken(StorageFile file)
+        {
+            var futureAccessList = StorageApplicationPermissions.FutureAccessList;
+            var existingEntries = futureAccessList.Entries;
+            var capacity = futureAccessList.MaximumItemsAllowed;
+            var now = DateTime.Now;
+
+            // Use more aggresive timeout when number of items exceed 100 or half of the capacity
+            // Note: 100 is an arbitary limit in case that OS reduces the capacity (it is 1000 in RS1). 
+            var expiry = existingEntries.Count >= capacity / 2 || existingEntries.Count >= 100 ?
+                now.AddDays(-1) :
+                now.AddDays(-7);
+
+            foreach (var entry in existingEntries)
+            {
+                if (DateTime.TryParse(entry.Metadata, out DateTime time) && time < expiry)
+                {
+                    futureAccessList.Remove(entry.Token);
+                }
+            }
+
+            return futureAccessList.Add(file, now.ToString(System.Globalization.CultureInfo.InvariantCulture));
+        }
+
+        private async void OnDragEnter(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+            var data = await GetDataTransferInfo(args.DataView);
+
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(new DragDropEvent(view.GetTag(), "topDragEnter", data));
+        }
+
+        private async void OnDragOver(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+
+            // TODO: Send this event to JS, get response (how?) and
+            // set the AcceptedOperation value. The JS handler in
+            // MessagePanel merely does this:
+            //
+            //      e.dataTransfer.effectAllowed = 'copy';
+            //      e.dataTransfer.dropEffect = 'copy';
```

Comment:
We should unify the drag and drop API in ReactXP to be the same between web, UWP, and the other platforms. For example, if we are going to hardcode "copy" into the UWP implementation, consider hardcoding it into the web implementation for consistency.

Reply:
I would add Xamarin.Mac and Linux into that list as well, since those are the main platforms we are targeting for the netstandard branch. WPF as well, if its not too deviant in its API.

Reply:
Is it worth doing this at this point? Hardcoding `copy` into the existing web implementation would be a breaking change. Instead, I think RNW will be heading the web direction: later there will be support for web-style event handlers that can modify the event object and send updates to the native part. The hardcoded `copy` is just the best possible solution in the current state of things.